source/application/AMPIC887DriverServer/AMPIC887ConsoleApplication.cpp

```diff
@@ -70,23 +66,507 @@ void AMPIC887ConsoleApplication::onActiveControllerChangeIssued(const QString &c
 	}
 }
 
-void AMPIC887ConsoleApplication::onGCS2CommandIssued(AMGCS2Command* command)
+void AMPIC887ConsoleApplication::onControllerOutputReceived(const QString &outputString)
 {
-	if(!controllerCollection_.activeController()) {
+	consoleInputHandler_->writeLineToStandardOutput(outputString);
+}
+
+void AMPIC887ConsoleApplication::onControllerErrorEncountered(const QString &errorMessage)
+{
+	consoleInputHandler_->writeLineToStandardError(errorMessage);
+}
+
+void AMPIC887ConsoleApplication::onSetCommandLevelCommandIssued(AMGCS2::ControllerCommandLevel commandLevel,
+																const QString &password)
+{
+	AMPIC887Controller* activeController = controllerCollection_.activeController();
+	if(!activeController) {
```

Comment:
Looking at all these setters (which all suffer from the same issue) I could also do with outputting an error message if no active controller was found. Would you be okay with:
```C++
AMPIC887Controller* activeController = controllerCollection_.activeController();

if(!activeController) {
    consoleInputHandler_->writeLineToStandardError("No active controller to issue command to.");
} else if(!activeController->setCommandLevel(commandLevel, password) {
    consoleInputHandler_->writeLineToStandardError(activeController->lastError()
}
```
???